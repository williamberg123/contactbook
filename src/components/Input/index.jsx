import PropTypes from 'prop-types';
import StyledInput from './styles';

export default function Input({ type, placeholder, elementRef }) {
	return (
		<StyledInput required ref={elementRef} type={type} placeholder={placeholder} />
	);
}

Input.propTypes = {
	type: PropTypes.string.isRequired,
	placeholder: PropTypes.string,
	elementRef: PropTypes.instanceOf(Object),
};
