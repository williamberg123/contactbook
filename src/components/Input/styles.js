import styled from 'styled-components';

const StyledInput = styled.input`
	padding: 15px;
`;

export default StyledInput;
