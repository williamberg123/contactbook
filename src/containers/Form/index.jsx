import PropTypes from 'prop-types';
import StyledForm from './styles';

export default function Form({ children, submitFunc }) {
	return (
		<StyledForm onSubmit={submitFunc}>
			{children}
		</StyledForm>
	);
}

Form.propTypes = {
	children: PropTypes.node.isRequired,
	submitFunc: PropTypes.func.isRequired,
};
