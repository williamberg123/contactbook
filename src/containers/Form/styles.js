import styled from 'styled-components';

const StyledForm = styled.form`
	max-width: 500px;
	margin: auto;
	padding: 20px;
	display: flex;
	flex-direction: column;
	box-shadow: 2px 2px 10px #999;
	border-radius: 10px;
`;

export default StyledForm;
