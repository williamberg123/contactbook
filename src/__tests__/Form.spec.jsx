import { fireEvent, render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import Form from '../containers/Form';
import Input from '../components/Input';

describe('<Form />', () => {
	it('should render Form children correctly', () => {
		render(
			<Form>
				<Input type="text" placeholder="text" />
				<Input type="number" placeholder="number" />
				<Input type="email" placeholder="email" />
			</Form>,
		);

		expect(screen.getByPlaceholderText('text')).toBeInTheDocument();
		expect(screen.getByPlaceholderText('number')).toBeInTheDocument();
		expect(screen.getByPlaceholderText('email')).toBeInTheDocument();
	});

	it('should call submit function when form is submited', () => {
		const fn = vi.fn();

		render(
			<Form submitFunc={fn}>
				<Input type="text" placeholder="text" />
			</Form>,
		);

		fireEvent.submit(screen.getByPlaceholderText('text').parentElement);

		expect(fn).toBeCalled();
	});

	it('should match snapshot', () => {
		render(
			<Form>
				<Input type="text" placeholder="text" />
				<Input type="number" placeholder="number" />
				<Input type="email" placeholder="email" />
			</Form>,
		);

		expect(screen.getByPlaceholderText('text').parentElement).toMatchSnapshot();
	});
});
