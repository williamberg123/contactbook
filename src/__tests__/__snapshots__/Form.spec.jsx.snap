// Vitest Snapshot v1

exports[`<Form /> > should match snapshot 1`] = `
.c0 {
  max-width: 500px;
  margin: auto;
  padding: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-shadow: 2px 2px 10px #999;
  border-radius: 10px;
}

.c1 {
  padding: 15px;
}

@media (max-width:600px) {
  .c1 {
    padding: 10px;
  }
}

<form
  class="c0"
>
  <input
    class="c1"
    placeholder="text"
    required=""
    type="text"
  />
  <input
    class="c1"
    placeholder="number"
    required=""
    type="number"
  />
  <input
    class="c1"
    placeholder="email"
    required=""
    type="email"
  />
</form>
`;
